src/include/function/string/operations/regexp_extract_all_operation.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include "common/types/ku_list.h"
+#include "common/vector/value_vector_utils.h"
+#include "re2.h"
+
+namespace kuzu {
+namespace function {
+namespace operation {
+
+struct RegexpExtractAll {
+    static inline void operation(common::ku_string_t& value, common::ku_string_t& pattern,
+        std::int64_t& group, common::list_entry_t& result, common::ValueVector& resultVector) {
+        std::vector<std::string> matches =
+            regexExtractAll(value.getAsString(), pattern.getAsString(), group);
+
+        result = common::ListVector::addList(&resultVector, matches.size());
+        auto resultValues = common::ListVector::getListValues(&resultVector, result);
+        auto resultDataVector = common::ListVector::getDataVector(&resultVector);
+        auto numBytesPerValue = resultDataVector->getNumBytesPerValue();
+
+        for (const auto& match : matches) {
+            common::ku_string_t kuString;
+            copyToKuzuString(match, kuString, *resultDataVector);
+            common::ValueVectorUtils::copyValue(resultValues, *resultDataVector,
+                reinterpret_cast<uint8_t*>(&kuString), *resultDataVector);
+            resultValues += numBytesPerValue;
+        }
+    }
+
+    static inline void operation(common::ku_string_t& value, common::ku_string_t& pattern,
+        common::list_entry_t& result, common::ValueVector& resultVector) {
+        int64_t defaultGroup = 0;
+        operation(value, pattern, defaultGroup, result, resultVector);
+    }
+
+    static std::vector<std::string> regexExtractAll(
+        const std::string& value, const std::string& pattern, std::int64_t& group) {
+        RE2 regex(common::parseCypherPatten(pattern));
+        auto submatchCount = regex.NumberOfCapturingGroups() + 1;
+        if (group > submatchCount) {
+            throw std::runtime_error("RE2: Match index is out of range");
```

Comment:
We have our own exception class, use common::RunTimeException instead of std::runtime_error.